### DISCUSSION TITLE
How do you setup Package Source Mapping for Uno.Check?

### DISCUSSION THREAD

**Discussion by MarkBrainfire** [unoplatform/uno#14457] [Q&A] [answered]
While cleaning up for the big switch to .NET 8 / VS 17.8 / Uno 5, I may have mistakenly deleted folders containing a manifest for Uno.Check??

Now when I attempt to install the latest version of Uno.Check I get the following error message:

``PS C:\Users\Mark> dotnet tool install -g uno.check --version 1.17.0 --add-source https://api.nuget.org/v3/index.json
Unhandled exception: Microsoft.DotNet.Cli.NuGetPackageDownloader.NuGetPackageInstallerException: Package Source Mapping is enabled, but no source found under the specified package ID: uno.check. See the documentation for Package Source Mapping at https://aka.ms/nuget-package-source-mapping for more details.
   at Microsoft.DotNet.Cli.NuGetPackageDownloader.NuGetPackageDownloader.LoadNuGetSources(PackageId packageId, PackageSourceLocation packageSourceLocation, PackageSourceMapping packageSourceMapping)
   at Microsoft.DotNet.Cli.NuGetPackageDownloader.NuGetPackageDownloader.GetBestPackageVersionAsync(PackageId packageId, VersionRange versionRange, PackageSourceLocation packageSourceLocation)
   at Microsoft.DotNet.Cli.ToolPackage.ToolPackageDownloader.<>c__DisplayClass8_0.<InstallPackage>b__0()
   at Microsoft.DotNet.Cli.TransactionalAction.Run[T](Func`1 action, Action commit, Action rollback)
   at Microsoft.DotNet.Tools.Tool.Install.ToolInstallGlobalOrToolPathCommand.Execute()
   at System.CommandLine.Invocation.InvocationPipeline.Invoke(ParseResult parseResult)
   at Microsoft.DotNet.Cli.Program.ProcessArgs(String[] args, TimeSpan startupTime, ITelemetry telemetryClient)``
   
How do you setup Package Source Mapping for Uno.Check?  Any help would be appreciated.

**Comment by MarkBrainfire** [comment, marked as answer]
SEE UPDATE AT THE END OF THIS COMMENT!

To solve this I created a nuget.config XML file in my c:\Users\Mark\\.NuGet\ directory.  In the path, note the period prior to the nuget folder name.  This directory is normally hidden in File Explorer, but can be viewed by checking the "Show hidden files, folders, and drives" option in File Explorer.  The nuget.config file created was as follows:

```
<?xml version="1.0" encoding="utf-8"?>
<configuration>
	<packageSourceMapping>
		<packageSource key="nuget.org">
			<package pattern="uno.check" />
		</packageSource>
	</packageSourceMapping>
	<packageSources>
		<add key="nuget.org" value="https://api.nuget.org/v3/index.json" />
	</packageSources>
</configuration>
```
Once this nuget.config xml file was created, it was necessary to refer to it when installing Uno.Check:

`dotnet tool install -g Uno.Check --configfile C:\Users\Mark\.nuget\nuget.config`

I was then able to successfully install Uno.Check and invoke it from anywhere using the Uno-Check command.

I ran Uno-Check, allowed it to repair the workloads, rebooted and ran it again.  I'm surprised that the workloads needed to be repaired?  After installing the newly released version of VS (17.8), I was able to successfully create Uno Platform .NET 7 solutions without problems.  During the Uno-Check "repair" I saw mention of a "Rollback file"?? 

**Now, even though Uno-Check reports that everything is properly setup, I am not able to create either a .NET 7 or .NET 8 Uno solution by customizing the Blank (left) wizard.**  

I'm running Uno-Check 1.17.0
VS 17.8
VS Uno Template 5.0.11.16

UPDATE
While the approach above was a useful exercise, you should NOT have to create a Package Source Mapping file in order to install Uno.Check.  After resetting my laptop to a clean version of Windows 11 / VS 17.8 etc. the problems that initiated this post were resolved.  See my reply to Jerome below.

**Reply by jeromelaban** [reply]
Thanks for the update. As always, when saying you are not able to something, make sure to provide logs or anything that could help us troubleshoot your issue.

**Reply by MarkBrainfire** [reply]
Yes I'll provide logs and binlogs.

**Reply by MarkBrainfire** [reply]
@jeromelaban and Uno friends,

I decided to go for the nuclear option and reset Windows 11 on my laptop.  This gave me the equivalent of a new, uncorrupted laptop.
I installed VS 17.8 with the required 3 workload groups adding C# WinAppSDK Templates to the .NET Desktop group.  This installed only .NET 8 on my machine.

Then I was able to install uno.check using the documented "dotnet tool install -g uno.check".  I allowed Uno-Check to enable long paths, install Open JDK 11, and update the Android and MacOS workloads.  

Then I was able, without any issues, to customize the blank wizard to create two solutions:

UnoWAX = .NET 8 / Windows, Android / XAML / CPM
and 
UnoWAMM = .NET 8 / Windows, Android / Markup / MAUI embedding, CPM

I'm headed to bed and expect to sleep well knowing that it appears the fundamentals are now in place to enable me to develop Brainfire, my education app that will change the world! 

... and I got a squeaky-clean new laptop in the process!

**Reply by jeromelaban** [reply]
Thanks a lot for the update! 

This may be a condition that was created somehow by the intermediate previews that were installed in between... That's sometimes the cost of being on the bleeding edge of .NET.
